### PR TITLE
feat: Update template for OIDC stack

### PR DESCRIPTION
**Pull request by Jimlinz**
This branch updates generic linz/template-python-hello-world git repository to suit oidc. Changes include:
- Removing support for Python 3.6
- Adding support for Python 3.10 
- Updating Poetry dependencies (incl. aws-cdk-lib which is needed for Python cdk)
- Removing files that are not relevant to this project

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -1,38 +0,0 @@
-name: "CodeQL"
```
Comment (l0b0) [inline .github/workflows/codeql-analysis.yml:1]: We might want to leave this. It's already functional, and it provides some measure of protection against secrets leaking, IIUC.

File: .gitignore [hunk 2]
```diff
@@ -2,8 +2,6 @@
 /html/
 /htmlcov/
```
Comment (l0b0) [inline .gitignore:3]: This is part of coverage, so it should probably go too.

File: .python-version [hunk 3]
```diff
@@ -1 +1 @@
-3.9.6
+3.10.4
```
Comment (l0b0) [inline .python-version:1]: This change probably belongs in a separate PR, no?

File: pyproject.toml [hunk 4]
```diff
@@ -34,22 +34,23 @@ authors = [
...
 
 [tool.poetry.dependencies]
-python = "^3.6"
+python = "^3.10"
 typer = "*"
+aws-cdk-lib = "^2.31.1"
 
 [tool.poetry.dev-dependencies]
 black = "*"
 coverage = "*"
```
Comment (l0b0) [inline pyproject.toml:50]: We can probably remove this package, `pytest*`, and the `[tool.coverage.run]` section above.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (linz-jim) [reply .github/workflows/codeql-analysis.yml]: Good point. Add this back in [8551140](https://github.com/linz/oidc-provider/pull/12/commits/85511406c43230450b086cad3d8dfa0f8c6da884)

Comment (linz-jim) [reply .github/workflows/codeql-analysis.yml]: CodeQL is failing at the moment due to
`Error: No code found during the build.`

I have not included any code in this branch. CDK changes are in [feat/cdk-stack](https://github.com/linz/oidc-provider/tree/feat/cdk-stack). Can we merge even if CodeQL fails?

File: .gitignore (hunk 2 above)
Comment (linz-jim) [reply .gitignore]: Removed in [dfd2bc2](https://github.com/linz/oidc-provider/pull/12/commits/dfd2bc28ad665aab0da623d90cd5804265bad765)

File: pyproject.toml (hunk 4 above)
Comment (linz-jim) [reply pyproject.toml]: `pytest` and `coverage` removed in [dfd2bc2](https://github.com/linz/oidc-provider/pull/12/commits/dfd2bc28ad665aab0da623d90cd5804265bad765)

However, I still see references to pytest in poetry.lock. Quick Google didn't yield anything useful. I tried `poetry lock --no-update` but seems to make no difference. 

I also ran `poetry update` to get `typer` and `jsii` to the latest version.

File: .python-version (hunk 3 above)
Comment (linz-jim) [reply .python-version]: `.python-version` file removed in [e74728c](https://github.com/linz/oidc-provider/pull/12/commits/e74728c5ad74d45a7a36c8a5239454e85a30a767)

I will update the documentation separately in the [fix/update-documentation](https://github.com/linz/oidc-provider/tree/fix/update-documentation) branch

File: pyproject.toml (hunk 4 above)
Comment (l0b0) [reply pyproject.toml]: > `aws-cdk-lib = "^2.31.1"` is needed for cdk stack no? (i.e. `from aws_cdk import Stack`)

GitHub comments are linked to a specific line, so I meant the "coverage" one.

Comment (l0b0) [reply pyproject.toml]: > However, I still see references to pytest in poetry.lock. Quick Google didn't yield anything useful. I tried `poetry lock --no-update` but seems to make no difference.

I wouldn't worry about it. That just means some other package has it as a dependency. You can find out by looking for `pytest` in `package.dependencies` in `poetry.lock`, or by checking `poetry show --tree`.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (l0b0) [reply .github/workflows/codeql-analysis.yml]: I'd rather not. Each branch should stand on its own as a single piece of work, with all its dependencies.
